addons/base_automation/models/base_automation.py

```diff
@@ -30,25 +30,97 @@
     False: 0,
 }
 
+CREATE_TRIGGERS = [
+    'on_create',
+
+    'on_create_or_write',
+    'on_priority_set',
+    'on_stage_set',
+    'on_state_set',
+    'on_tag_set',
+    'on_user_set',
+]
+
+WRITE_TRIGGERS = [
+    'on_write',
+    'on_archive',
+    'on_unarchive',
+
+    'on_create_or_write',
+    'on_priority_set',
+    'on_stage_set',
+    'on_state_set',
+    'on_tag_set',
+    'on_user_set',
+]
+
+CREATE_WRITE_SET = set(CREATE_TRIGGERS + WRITE_TRIGGERS)
+
+TIME_TRIGGERS = [
+    'on_time',
+    'on_time_created',
+    'on_time_updated',
+]
+
 
 class BaseAutomation(models.Model):
     _name = 'base.automation'
-    _description = 'Automated Action'
-    _order = 'sequence'
+    _description = 'Automation Rule'
 
-    action_server_id = fields.Many2one(
-        'ir.actions.server', 'Server Actions',
-        domain="[('model_id', '=', model_id)]",
-        delegate=True, required=True, ondelete='restrict')
+    name = fields.Char(string="Automation Rule Name", required=True, translate=True)
+    model_id = fields.Many2one(
+        "ir.model", string="Model", required=True, ondelete="cascade", help="Model on which the automation rule runs."
+    )
+    model_name = fields.Char(related="model_id.model", string="Model Name", readonly=True)
+    action_server_ids = fields.One2many("ir.actions.server", "base_automation_id",
```

Comment:
Wouldn't you like a `Many2many` instead ?

That would:
- functionally, allow users to use the same server action in 2 different automation, rather than forcing them to create 2 different server action when they want to execute the same code for 2 different automations,
- technically, avoid the need to add the Many2one field `base_automation_id` on `ir.server.actions`

Reply:
Then we would be back to the problem for which we introduced a link model initially: sequencing (and more importantly _reseauencing_) would not work.

But order of actions is import: if I set my automation to add a follower then send an email, and it somehow gets inverted, it's a problem.

Reply:
Ok I understand